compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/NameCollision.hs

```diff
@@ -24,6 +25,13 @@ import Control.Monad.Except (throwError)
 -- see 'FRName'.
 data Name
     = NModule ModuleName
+    | NVirtualModule ModuleName ModuleName
+    -- ^ For a module A.B.C we insert virtual module names A and A.B.
+    -- This is an extension of the check that prevents you from having
+    -- a type B in module A if you have a module A.B to also fail if you
+    -- have a module A.B.C but not A.B. At the moment, we only emit a warning
```

Comment:
I completely misunderstood this comment when first reading. I read it as: We want to enforce that `A.B` exists whenever `A.B.C` exists. I suggest we slightly reword this to something like:

> This is used to check that you do not have a type `B` in a module `A` and a module `A.B.C` at the same time, even if you don't have a module `A.B`.

Reply:
Thanks, copied your wording!